sycl/source/detail/bindless_images.cpp

```diff
@@ -22,37 +22,34 @@ namespace sycl {
 inline namespace _V1 {
 namespace ext::oneapi::experimental {
 
-void populate_pi_structs(const image_descriptor &desc, pi_image_desc &piDesc,
-                         pi_image_format &piFormat, size_t pitch = 0) {
-  piDesc = {};
-  piDesc.image_width = desc.width;
-  piDesc.image_height = desc.height;
-  piDesc.image_depth = desc.depth;
+void populate_ur_structs(const image_descriptor &desc, ur_image_desc_t &urDesc,
+                         ur_image_format_t &urFormat, size_t pitch = 0) {
+  urDesc = {};
+  urDesc.width = desc.width;
+  urDesc.height = desc.height;
+  urDesc.depth = desc.depth;
 
   if (desc.array_size > 1) {
     // Image array or cubemap
-    piDesc.image_type = desc.type == image_type::cubemap
-                            ? PI_MEM_TYPE_IMAGE_CUBEMAP
-                        : desc.height > 0 ? PI_MEM_TYPE_IMAGE2D_ARRAY
-                                          : PI_MEM_TYPE_IMAGE1D_ARRAY;
+    urDesc.type = desc.type == image_type::cubemap
+                      ? UR_MEM_TYPE_IMAGE_CUBEMAP_EXP
+                  : desc.height > 0 ? UR_MEM_TYPE_IMAGE2D_ARRAY
+                                    : UR_MEM_TYPE_IMAGE1D_ARRAY;
   } else {
-    piDesc.image_type =
-        desc.depth > 0
-            ? PI_MEM_TYPE_IMAGE3D
-            : (desc.height > 0 ? PI_MEM_TYPE_IMAGE2D : PI_MEM_TYPE_IMAGE1D);
+    urDesc.type = desc.depth > 0 ? UR_MEM_TYPE_IMAGE3D
+                                 : (desc.height > 0 ? UR_MEM_TYPE_IMAGE2D
+                                                    : UR_MEM_TYPE_IMAGE1D);
   }
 
-  piDesc.image_row_pitch = pitch;
-  piDesc.image_array_size = desc.array_size;
-  piDesc.image_slice_pitch = 0;
-  piDesc.num_mip_levels = desc.num_levels;
-  piDesc.num_samples = 0;
-  piDesc.buffer = nullptr;
-
-  piFormat = {};
-  piFormat.image_channel_data_type =
-      sycl::detail::convertChannelType(desc.channel_type);
-  piFormat.image_channel_order = sycl::detail::convertChannelOrder(
+  urDesc.rowPitch = pitch;
+  urDesc.arraySize = desc.array_size;
+  urDesc.slicePitch = 0;
+  urDesc.numMipLevel = desc.num_levels;
+  urDesc.numSamples = 0;
```

Comment:
I see no buffer=nullptr and suppose it means here pointer to extension-specific structure. urDesc.pNext seems to be representing this and should be initialized appropriately here.

Reply:
`pNext` will get initialized to `nullptr` with the `= {};` [here](https://github.com/intel/llvm/pull/14145/files#diff-3d5ac2d9d0d6a73eeeaa9d594baf7719e7f1b2a38785837707228eec3002951aR26), is that sufficient?